pinot-core/src/main/java/org/apache/pinot/core/operator/streaming/StreamingInstanceResponseOperator.java

```diff
@@ -34,25 +41,53 @@
 public class StreamingInstanceResponseOperator extends InstanceResponseOperator {
 
   private final StreamObserver<Server.ServerResponse> _streamObserver;
+  private final int _limit;
+  private int _numRowsCollected;
 
   public StreamingInstanceResponseOperator(BaseCombineOperator<?> combinedOperator, List<IndexSegment> indexSegments,
       List<FetchContext> fetchContexts, StreamObserver<Server.ServerResponse> streamObserver,
       QueryContext queryContext) {
     super(combinedOperator, indexSegments, fetchContexts, queryContext);
     _streamObserver = streamObserver;
+    _limit = queryContext.getLimit();
+    _numRowsCollected = 0;
   }
 
   @Override
   protected InstanceResponseBlock getNextBlock() {
-    InstanceResponseBlock responseBlock = super.getNextBlock();
-    InstanceResponseBlock metadataOnlyResponseBlock = responseBlock.toMetadataOnlyResponseBlock();
+    prefetchAll();
+    InstanceResponseBlock exceptionResultBlock = null;
+    BaseResultsBlock combinedResult;
     try {
-      _streamObserver.onNext(StreamingResponseUtils.getDataResponse(responseBlock.toDataOnlyDataTable()));
+      combinedResult = _combineOperator.nextBlock();
+      while (!(combinedResult instanceof MetadataResultsBlock)) {
+        if (combinedResult instanceof ExceptionResultsBlock) {
+          exceptionResultBlock = new InstanceResponseBlock(combinedResult, _queryContext);
+          return exceptionResultBlock;
+        } else {
+          sendBlock(combinedResult);
+        }
+        combinedResult = _combineOperator.nextBlock();
+      }
     } catch (IOException e) {
-      metadataOnlyResponseBlock.addException(
+      exceptionResultBlock = new InstanceResponseBlock();
+      exceptionResultBlock.addException(
           QueryException.getException(QueryException.DATA_TABLE_SERIALIZATION_ERROR, e));
+      return exceptionResultBlock;
+    } finally {
+      releaseAll();
```

Comment:
We can clean up the combine operator sub-tasks here. No need to add clean up everywhere in the combine operator

Reply:
looks like moving this to instance operator makes lots of combine operator tests to fail will need another round of clean up

Reply:
clean up done